serve_diffusion_component.py

```diff
@@ -1,24 +1,50 @@
-# ! pip install torch diffusers git+https://github.com/Lightning-AI/lightning-diffusion-component.git
-import diffusers
+# !pip install 'git+https://github.com/Lightning-AI/LAI-API-Access-UI-Component.git@diffusion'
+# !pip install 'git+https://github.com/Lightning-AI/stablediffusion.git@lit'
+# !curl https://raw.githubusercontent.com/runwayml/stable-diffusion/main/configs/stable-diffusion/v1-inference.yaml -o v1-inference.yaml
+
+
 import lightning as L
-import torch
+import base64, io, os, torch, lightning_diffusion
+from ldm.lightning import LightningStableDiffusion, PromptDataset
 
-from lightning_diffusion import BaseDiffusion, download_from_lightning_cloud
 
+class ServeDiffusion(lightning_diffusion.BaseDiffusion):
+    def __init__(self, input_type=L.app.components.Text, output_type=L.app.components.Image, **kwargs):
+        super().__init__(input_type=input_type, output_type=output_type, **kwargs)
+        self._model = None
 
-class ServeDiffusion(BaseDiffusion):
-    def setup(self, *args, **kwargs):
-        download_from_lightning_cloud(
-            "daniela/stable_diffusion", version="latest", output_dir="model"
+    def setup(self):
+        os.system(
+            "curl -C - https://pl-public-data.s3.amazonaws.com/dream_stable_diffusion/v1-5-pruned-emaonly.ckpt -o v1-5-pruned-emaonly.ckpt"
         )
-        self.model = diffusers.StableDiffusionPipeline.from_pretrained("model").to(
-            self.device
+        self._trainer = L.Trainer(
+            accelerator="auto",
+            devices=1,
+            precision=16 if torch.cuda.is_available() else 32,
+            enable_progress_bar=False,
+            inference_mode=False,
         )
+        self._model = LightningStableDiffusion(
+            config_path="v1-inference.yaml",
+            checkpoint_path="v1-5-pruned-emaonly.ckpt",
+            device=self._trainer.strategy.root_device.type,
+            size=512,
+        )
+        if torch.cuda.is_available():
+            self._model = self._model.to(torch.float16)
+            torch.cuda.empty_cache()
+
 
-    def predict(self, data):
-        with torch.autocast(device_type=self.device.type, dtype=torch.float16):
-            out = self.model(prompt=data.prompt, num_inference_steps=23)
-            return {"image": self.serialize(out[0][0])}
+    def predict(self, request):
+        with torch.no_grad():
```

Comment:
There is not need to use  `with torch.no_grad()` if using the latest version of lightning,   removing this cam help to make the code simpler?